src/System.Runtime.Extensions/tests/System/BitConverter.cs

```diff
@@ -115,6 +115,22 @@ public static void DoubleToInt64Bits()
             Assert.Equal(input, roundtripped);
         }
 
+		[Fact]
+		public static void SingleNaNToInt32Bits()
+		{
+			int bits = BitConverter.SingleToInt32Bits(float.NaN);
+			Assert.Equal (0x7f800000, bits & 0x7f800000);
```

Comment:
Why not just assert that `(bits & 0x7FFFFFFF) > 0x7FC00000` (which is basically `IsQuietNaN`)?

Reply:
I didn't want to copy/paste it 😄 but that isn't a good reason, I'll change it, as it's easier to read.

Any reason why you suggest to restrict it to `0x7fc00000` (which will catch quiet NaNs only)? `0x7f800000` would catch all representations of NaN. Is it expected that `float.NaN` is _always_ a quiet NaN?

Reply:
> Is it expected that float.NaN is always a quiet NaN?

Yes.

Reply:
Thanks, updated.

Reply:
Tiny correction: It must be `(bits & 0x7FFFFFFF) >= 0x7FC00000`.

Reply:
This relaxation of the check would make sense for value computed at runtime. For example, if we had:

```
static double x = 0.0;

static void Test()
{
    double y = x/x;
    ... it is fine to assert that y matches any valid NaN pattern ...
}
```

However, this relaxation does not make sense to me for validation of a NaN constant. It is working around Roslyn determinism bug - Roslyn is expected to produce a deterministic output.